test/integration/targets/lookup_config/tasks/main.yml

```diff
@@ -0,0 +1,59 @@
+- name: Verify lookup_config errors with no on_missing (failure expected)
+  set_fact:
+    foo: '{{lookup("config", "THIS_DOES_NOT_EXIST")}}'
+  ignore_errors: yes
+  register: lookup_config_1
+
+- name: Verify lookup_config errors with on_missing=error (failure expected)
+  set_fact:
+    foo: '{{lookup("config", "THIS_DOES_NOT_EXIST", on_missing="error")}}'
+  ignore_errors: yes
+  register: lookup_config_2
+
+- name: Verify lookup_config errors with on_missing=skip
+  set_fact:
+    lookup3: '{{lookup("config", "THIS_DOES_NOT_EXIST", on_missing="skip")}}'
+  register: lookup_config_3
+
+# TODO: Is there a decent way to check that the warning is actually triggered?
```

Comment:
Currently any tests that need to consult warnings do so by running as a script-based test, saving the playbook output and then searching it for the desired output.